mlflow/catboost.py

```diff
@@ -0,0 +1,303 @@
+
+    :param kwargs: kwargs to pass to `CatBoost.save_model`_ method.
+    """
+    Model.log(
+        artifact_path=artifact_path,
+        flavor=mlflow.catboost,
+        registered_model_name=registered_model_name,
+        cb_model=cb_model,
+        conda_env=conda_env,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+        **kwargs,
+    )
+
+
+def _init_model(model_type):
+    from catboost import CatBoost, CatBoostClassifier, CatBoostRegressor
+
+    model_types = {c.__name__: c for c in [CatBoost, CatBoostClassifier, CatBoostRegressor]}
+
+    if model_type not in model_types:
+        raise TypeError(
+            "Invalid model type: '{}'. Must be one of {}".format(
+                model_type, list(model_types.keys())
+            )
+        )
+
+    return model_types[model_type]()
+
+
+def _load_model(path, model_type, save_format):
+    model = _init_model(model_type)
+    model.load_model(os.path.abspath(path), save_format)
+    return model
+
+
+def _load_pyfunc(path):
+    """
+    Load PyFunc implementation. Called by ``pyfunc.load_pyfunc``.
+
+    :param path: Local filesystem path to the MLflow Model with the ``catboost`` flavor.
+    """
+    flavor_conf = _get_flavor_configuration(
+        model_path=os.path.dirname(path), flavor_name=FLAVOR_NAME
+    )
+    return _CatboostModelWrapper(
+        _load_model(path, flavor_conf.get(_MODEL_TYPE_KEY), flavor_conf.get(_SAVE_FORMAT_KEY))
+    )
+
+
+def load_model(model_uri):
+    """
+    Load a CatBoost model from a local file or a run.
+
+    :param model_uri: The location, in URI format, of the MLflow model. For example:
+
+                      - ``/Users/me/path/to/local/model``
+                      - ``relative/path/to/local/model``
+                      - ``s3://my_bucket/path/to/model``
+                      - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+
+                      For more information about supported URI schemes, see
+                      `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
+                      artifact-locations>`_.
+
+    :return: A CatBoost model (an instance of `CatBoost`_, `CatBoostClassifier`_,
+             or `CatBoostRegressor`_)
+    """
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri)
+    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
+    cb_model_file_path = os.path.join(
+        local_model_path, flavor_conf.get(_MODEL_BINARY_KEY, _MODEL_BINARY_FILE_NAME)
+    )
+    return _load_model(
+        cb_model_file_path, flavor_conf.get(_MODEL_TYPE_KEY), flavor_conf.get(_SAVE_FORMAT_KEY)
+    )
+
+
+class _CatboostModelWrapper:
+    def __init__(self, cb_model):
+        self.cb_model = cb_model
+
+    def predict(self, dataframe):
+        return self.cb_model.predict(dataframe)
```

Comment:
The `cb_model.predict` provides many arguments, could we support passing those arguments ?
xgboost flavor has similar issue.

Reply:
Good point, but `PyfuncModel` currently don't support propagating optional arguments to the wrapped model.